packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -73,4 +73,69 @@ export class ApplicationExtension<
     public beforeRouteMatch(routes: RouteProps[]): RouteProps[] {
         return routes
     }
+
+    public async serialize(): Promise<SerializedRoute[]> {
+        const routes = await this.getRoutes()
+        return routes.map((route) => {
+            if (!route.component?.displayName) {
+                throw new Error(`Component for route with path "${route.path}" is missing a displayName`)
+            }
+            return {
+                path: route.path,
+                componentName: route.component?.displayName,
+                // TODO: do we need to serialize props?
+                //componentProps: route.props
+            }
+        })
+    }
+
+    public deserialize(serializedRoutes: SerializedRoute[], componentMap: Modules): any[] {
```

Comment:
Currently I pass in the `componentMap`, I'll look to refactor this code to call the `this.getComponentMap` directly